sound/soc/intel/boards/Kconfig

```diff
@@ -431,7 +431,7 @@ config SND_SOC_INTEL_SOF_RT5682_MACH
 		   (SND_SOC_SOF_BAYTRAIL && (X86_INTEL_LPSS || COMPILE_TEST))
 	select SND_SOC_RT5682
 	select SND_SOC_DMIC
-	select SND_SOC_HDAC_HDMI
+	select SND_SOC_HDAC_HDMI if SND_SOC_SOF_HDA_LINK
```

Comment:
this one makes no sense, or the fix is in the wrong position. The code in the machine drivers does not have conditional compilation support, so if you don't have SOF_HDA_LINK things will be broken.

I think the right fix would be to change line 426 as well:
if SND_SOC_SOF_BAYTRAIL || SND_SOC_SOF_HDA_LINK

@bardliao what do you think?

Reply:
Agree with @plbossart  We should always select SND_SOC_HDAC_HDMI if we want to use sof_rt5682 machine driver.

Reply:
@plbossart OK. I will make the patch as what you suggested.